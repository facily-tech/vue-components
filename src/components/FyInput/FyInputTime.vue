<template>
  <fy-input
    :value="text"
    @input="update"
    v-mask="mask"
    :rules="timeRule"
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </fy-input>
</template>

<script lang="ts">
import Vue from 'vue';
import { VueMaskDirective } from 'v-mask';
import FyInput from './FyInput.vue';

export default Vue.extend({
  name: 'fy-input-time',

  directives: { mask: VueMaskDirective },

  components: {
    FyInput,
  },

  props: {
    value: {
      type: String,
    },
    timeRule: {
      type: Array,
      default: () => [
        (value: string): boolean | string => {
          const nValue = value.split(':');

          let msg = '';

          if (Number(nValue[0]) > 23) msg = 'Hora inválida';

          if (Number(nValue[1]) > 59) msg = 'Minutos inválidos';

          if (Number(nValue[0]) > 31 && Number(nValue[1]) > 12) msg = 'Hora e minitos inválidos';

          if (!msg) return true;

          return msg;
        },
      ],
    },
  },

  data() {
    return {
      text: this.value ? this.value : '',
      mask: '##:##',
    };
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },

  methods: {
    update(value: string): void {
      this.$emit('input', value);
    },
  },
});
</script>
