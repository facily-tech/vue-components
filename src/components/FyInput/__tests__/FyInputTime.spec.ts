import Vue from 'vue';
import Vuetify from 'vuetify';
import FyInputTime from '../FyInputTime.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyInputTime.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyInputTime>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyInputTime, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
