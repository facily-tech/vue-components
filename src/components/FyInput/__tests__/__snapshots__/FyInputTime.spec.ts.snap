// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyInputTime.ts should render component and match snapshot 1`] = `
<div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined" value="" rules="(value) => {
            /* istanbul ignore next */
            cov_jcetl8oz9.f[2]++;
            const nValue =
            /* istanbul ignore next */
            (cov_jcetl8oz9.s[9]++, value.split(':'));
            let msg =
            /* istanbul ignore next */
            (cov_jcetl8oz9.s[10]++, '');

            /* istanbul ignore next */
            cov_jcetl8oz9.s[11]++;

            if (Number(nValue[0]) > 23) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[0][0]++;
              cov_jcetl8oz9.s[12]++;
              msg = 'Hora inválida';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[0][1]++;
            }

            cov_jcetl8oz9.s[13]++;

            if (Number(nValue[1]) > 59) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[1][0]++;
              cov_jcetl8oz9.s[14]++;
              msg = 'Minutos inválidos';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[1][1]++;
            }

            cov_jcetl8oz9.s[15]++;

            if (
            /* istanbul ignore next */
            (cov_jcetl8oz9.b[3][0]++, Number(nValue[0]) > 31) &amp;&amp;
            /* istanbul ignore next */
            (cov_jcetl8oz9.b[3][1]++, Number(nValue[1]) > 12)) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[2][0]++;
              cov_jcetl8oz9.s[16]++;
              msg = 'Hora e minitos inválidos';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[2][1]++;
            }

            cov_jcetl8oz9.s[17]++;

            if (!msg) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[4][0]++;
              cov_jcetl8oz9.s[18]++;
              return true;
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[4][1]++;
            }

            cov_jcetl8oz9.s[19]++;
            return msg;
          }" timerule="(value) => {
            /* istanbul ignore next */
            cov_jcetl8oz9.f[2]++;
            const nValue =
            /* istanbul ignore next */
            (cov_jcetl8oz9.s[9]++, value.split(':'));
            let msg =
            /* istanbul ignore next */
            (cov_jcetl8oz9.s[10]++, '');

            /* istanbul ignore next */
            cov_jcetl8oz9.s[11]++;

            if (Number(nValue[0]) > 23) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[0][0]++;
              cov_jcetl8oz9.s[12]++;
              msg = 'Hora inválida';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[0][1]++;
            }

            cov_jcetl8oz9.s[13]++;

            if (Number(nValue[1]) > 59) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[1][0]++;
              cov_jcetl8oz9.s[14]++;
              msg = 'Minutos inválidos';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[1][1]++;
            }

            cov_jcetl8oz9.s[15]++;

            if (
            /* istanbul ignore next */
            (cov_jcetl8oz9.b[3][0]++, Number(nValue[0]) > 31) &amp;&amp;
            /* istanbul ignore next */
            (cov_jcetl8oz9.b[3][1]++, Number(nValue[1]) > 12)) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[2][0]++;
              cov_jcetl8oz9.s[16]++;
              msg = 'Hora e minitos inválidos';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[2][1]++;
            }

            cov_jcetl8oz9.s[17]++;

            if (!msg) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[4][0]++;
              cov_jcetl8oz9.s[18]++;
              return true;
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[4][1]++;
            }

            cov_jcetl8oz9.s[19]++;
            return msg;
          }">
  <div class="v-input__control">
    <div class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">​</span></legend>
      </fieldset>
      <div class="v-text-field__slot"><input data-test="input" timerule="(value) => {
            /* istanbul ignore next */
            cov_jcetl8oz9.f[2]++;
            const nValue =
            /* istanbul ignore next */
            (cov_jcetl8oz9.s[9]++, value.split(':'));
            let msg =
            /* istanbul ignore next */
            (cov_jcetl8oz9.s[10]++, '');

            /* istanbul ignore next */
            cov_jcetl8oz9.s[11]++;

            if (Number(nValue[0]) > 23) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[0][0]++;
              cov_jcetl8oz9.s[12]++;
              msg = 'Hora inválida';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[0][1]++;
            }

            cov_jcetl8oz9.s[13]++;

            if (Number(nValue[1]) > 59) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[1][0]++;
              cov_jcetl8oz9.s[14]++;
              msg = 'Minutos inválidos';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[1][1]++;
            }

            cov_jcetl8oz9.s[15]++;

            if (
            /* istanbul ignore next */
            (cov_jcetl8oz9.b[3][0]++, Number(nValue[0]) > 31) &amp;&amp;
            /* istanbul ignore next */
            (cov_jcetl8oz9.b[3][1]++, Number(nValue[1]) > 12)) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[2][0]++;
              cov_jcetl8oz9.s[16]++;
              msg = 'Hora e minitos inválidos';
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[2][1]++;
            }

            cov_jcetl8oz9.s[17]++;

            if (!msg) {
              /* istanbul ignore next */
              cov_jcetl8oz9.b[4][0]++;
              cov_jcetl8oz9.s[18]++;
              return true;
            } else
            /* istanbul ignore next */
            {
              cov_jcetl8oz9.b[4][1]++;
            }

            cov_jcetl8oz9.s[19]++;
            return msg;
          }" id="input-3" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
